lightcurve/static/lightcurve/js/lightcurve.js

```diff
@@ -61,4 +73,19 @@ $(function() {
     window.location.reload();
     return false;
   });
+
+  $(document).on('click', '.edit-name', function(e) {
+    e.preventDefault();
+    var filename = $(this).prev(".lightcurve-name").text();
+    $( $(this).prev(".lightcurve-name") ).replaceWith( "<input type='name' class='input_name' name='input_name' value='" + filename + "'>" );
```

Comment:
I think it would be cleaner to put these buttons in the HTML with `style="display:none"` and show/hide them accordingly.  The rationale for this is that if we build HTML in one place, rather than in template and in scripts, it's easier to find, read, and understand how the app behaves.

Then, you can just bind an event to the Save button and handle the `$.post` there rather than using a form submit event.

Reply:
Done with the "style='display:none'" part. For the "bind an event" part, does that mean instead of `onsubmit='saveLightcurveChanges({{lightcurve.id}}); return false;'`, I should do `$(form).on('submit')` instead?

Reply:
Yes, and I also think you should give the form a class `js-edit-name-form` or something like that so things don't get weird if we add another form on the page.